src/api/filter.rs

```diff
@@ -0,0 +1,122 @@
+use super::handler::{add_order, get_orders};
+use crate::models::Order;
+use crate::models::OrderBook;
+use warp::Filter;
+
+fn json_body() -> impl Filter<Extract = (Order,), Error = warp::Rejection> + Clone {
+    // When accepting a body, we want a JSON body
+    // (and to reject huge payloads)...
+    warp::body::content_length_limit(1024 * 16).and(warp::body::json())
+}
+
+fn with_orderbook(
+    orderbook: OrderBook,
+) -> impl Filter<Extract = (OrderBook,), Error = std::convert::Infallible> + Clone {
+    warp::any().map(move || orderbook.clone())
+}
+
+pub fn post_order(
+    orderbook: OrderBook,
+) -> impl Filter<Extract = impl warp::Reply, Error = warp::Rejection> + Clone {
+    warp::post()
+        .and(warp::path("v1"))
+        .and(warp::path("orders"))
+        .and(warp::path::end())
+        .and(json_body())
+        .and(with_orderbook(orderbook))
+        .and_then(add_order)
+}
+
+pub fn get(
+    orderbook: OrderBook,
+) -> impl Filter<Extract = impl warp::Reply, Error = warp::Rejection> + Clone {
+    warp::get()
+        .and(warp::path("v1"))
+        .and(warp::path("orders"))
+        .and(warp::path::end())
+        .and(warp::path::end())
+        .and(with_orderbook(orderbook))
+        .and_then(get_orders)
+}
+
+#[cfg(test)]
+pub mod test_util {
+    use super::*;
+    use crate::models::Order;
+    use ethcontract::web3::types::U256;
+    use warp::http::StatusCode;
+    use warp::test::request;
+
+    #[tokio::test]
```

Comment:
Instead of tokio tests, can we use `now_or_never` like in our price estimator [tests](https://github.com/gnosis/dex-services/blob/7c7237b52d1d5614d73935d84fb706fd40f4c3f6/price-estimator/src/filter.rs#L443-L450)?

Reply:
Personally, I prefer `tokio::test`. It allows test code to use `async` directly which is very nice.